other/tests.py

```diff
@@ -199,6 +202,67 @@ def test_news_notifications(self):
         self.assertEqual(len(response.data), 1)
         self.assertEqual(response.data[0]['actor']['title'], ne1.title)
 
+    def test_comment_notifications(self):
+        """Test notifications for complaint subscribers when comments are made"""
+        # Creating dummy complaint with one subscriber (the creator)
+        test_creator = get_new_user()
+        complaint = Complaints.objects.create(created_by=test_creator.profile)
+        complaint.subscriptions.add(test_creator.profile)
+
+        # Comments url
+        url = '/api/venter/complaints/' + str(complaint.id) + '/comments'
+
+        # First comment on the new complaint, should generate 1 new notification (Total = 1)
+        test_commenter_1 = User.objects.create_user(username='Commenter1', password='Commenter1@123')
+        UserProfile.objects.create(name='CommentProfile1', user=test_commenter_1)
+        self.client.force_authenticate(user=test_commenter_1)
+
+        data = {'text': 'test'}
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(response.status_code, 201)
+
+        # Get notifications for test_creator.
+        # There should be one notification from the first comment made by test_commenter_1
+        url = '/api/notifications'
+        self.client.force_authenticate(user=test_creator)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['actor']['commented_by']['id'], str(test_commenter_1.profile.id))
+
+        # Comment url
+        url = '/api/venter/complaints/' + str(complaint.id) + '/comments'
+
+        # Second comment on the complaint, should generate 2 new notifications.
+        # One new notification each will be made for test_creator and test_commenter_1.
+        # Both will be tested independently.
+        test_commenter_2 = User.objects.create_user(username='Commenter2', password='Commenter2@123')
```

Comment:
@samjakes why are you using this method? There's a helper function to create new users.

Reply:
I'll try using the helper function. I'm not sure if the users themselves are unique or easy to reference for the notifications function though. I'm running separate tests to check for notifications received by separate users. For this, I have to force authenticate my users separately when I'm making GET requests to the notifications api. I wrote this in so that I could keep better track of the users, but I'll try using the helper functions and revert to you with the results